src/components/memorygame/cardInitialStateGenerator.jsx

```diff
@@ -0,0 +1,65 @@
+import * as invariant from "invariant";
+import CARD_BACKSIDE_IMAGES from "./memoryCardBackSideImages";
+import shuffle from "./shuffle";
+
+/**
+ * cardDataArr example:
+ * [{description:"...",link:"...",phrase:".."},{description:"...",link:"...",phrase:".."},...]
```

Comment:
nicely done. if we were using typescript, we wouldn't need to do this because the types would be clearly declared, but in javascript we have to do things like this. this is also not ideal because these comments don't age well in case you change the structure but there's not much that can be done about that